api-server/src/main/java/vermillion/http/HttpServerVerticle.java

```diff
@@ -44,1048 +44,1139 @@
+		} catch (Exception e) {
+			apiFailure(context, new BadRequestThrowable("Body is not a valid JSON"));
+			return;
+		}
+
+		logger.debug("Body=" + requestBody.encode());
+
+		if (!requestBody.containsKey("id")) {
+			apiFailure(context, new BadRequestThrowable("No id found in body"));
+			return;
+		}
+
+		if (!requestBody.containsKey("geo_distance")
+				&& !requestBody.containsKey("time")
+				&& !requestBody.containsKey("attribute")) {
+			apiFailure(context, new BadRequestThrowable("Invalid request"));
+			return;
+		}
+
+		Object resourceIdObj = requestBody.getValue("id");
+
+		if (!(resourceIdObj instanceof String) && !(resourceIdObj instanceof JsonArray)) {
+			apiFailure(context, new BadRequestThrowable("Resource id is not valid"));
+			return;
+		}
+		if (resourceIdObj instanceof JsonArray) {
+			resourceIDArray = requestBody.getJsonArray("id");
+			for (Object o : resourceIDArray) {
+				if (!(o instanceof String)) {
+					apiFailure(
+							context, new BadRequestThrowable("Resource ID list should be a list of strings"));
+					return;
+				}
+				if ("".equalsIgnoreCase(o.toString())) {
+					apiFailure(context, new BadRequestThrowable("Resource ID is empty"));
+					return;
+				}
+				if (!(((String) o).endsWith(".public")) && token == null) {
+					apiFailure(context, new BadRequestThrowable("No token found in request"));
+					return;
+				}
+			}
+			termsQuery.getJsonObject("terms").put("id.keyword", resourceIDArray);
+			filterQuery.add(termsQuery);
+		} else {
+			resourceIDstr = requestBody.getString("id");
+			if ("".equalsIgnoreCase(resourceIDstr)) {
+				apiFailure(context, new BadRequestThrowable("Resource ID is empty"));
+				return;
+			}
+			if (!resourceIDstr.endsWith(".public") && token == null) {
+				apiFailure(context, new BadRequestThrowable("No token found in request"));
+				return;
+			}
+			termQuery.getJsonObject("term").put("id.keyword", resourceIDstr);
+			filterQuery.add(termQuery);
+		}
+
+		// Geo Query
+		if (requestBody.containsKey("geo_distance")) {
+			Object geoDistanceObj = requestBody.getValue("geo_distance");
+
+			if (!(geoDistanceObj instanceof JsonObject)) {
+				apiFailure(context, new BadRequestThrowable("Geo distance is not a valid Json Object"));
+				return;
+			}
+
+			JsonObject geoDistance = requestBody.getJsonObject("geo_distance");
+
+			logger.debug("geo distance=" + geoDistance.encodePrettily());
+
+			if (!geoDistance.containsKey("coordinates") || !geoDistance.containsKey("distance")) {
+				apiFailure(
+						context,
+						new BadRequestThrowable("Geo distance does not contain coordinates and/or distance"));
+				return;
+			}
+
+			Object distanceObj = geoDistance.getValue("distance");
+
+			if (!(distanceObj instanceof String)) {
+				apiFailure(context, new BadRequestThrowable("Distance is not a string"));
+				return;
+			}
+			String distance = geoDistance.getString("distance");
+
+			if (!distance.endsWith("m") && !distance.endsWith("M")) {
```

Comment:
I can spot a problem here. If the units are specified as ``cm``, ``mm`` etc., the results are still in metres. The ideal response should be that anything apart from metres is not supported.

Reply:
Why is there no test case covering this? @vasanth096

Reply:
Added Tests